factionRenderer.py

```diff
@@ -1,105 +1,116 @@
+
+def render_factionstore(faction, items):
+    faction_name = faction[:-5]
+    results_filename = faction_name+"_store_Table.wiki"
+    faction_info=get_faction_specific_info(faction_name)
+    weapons = []
+    ammunitions = []
+    gears = []
+    mechs = []
+    vehicles = []
+    battlearmors = []
+    contracts = []
+    for item in items:
+        if item.startswith("Weapon_"):
+            weapons.append(item)
+        elif item.startswith("Ammo_"):
+            ammunitions.append(item)
+        elif item.startswith(("Gear_", "emod_")) and not item.startswith("Gear_Contract_"):
+            gears.append(item)
+        elif item.startswith("mechdef_") and not item.startswith("mechdef_ba"):
+            mechs.append(item)
+        elif item.startswith("vehicledef_"):
+            vehicles.append(item)
+        elif item.startswith("mechdef_ba"):
+            battlearmors.append(item)
+        elif item.startswith("Gear_Contract_"):
+            contracts.append(item)
+    
+    for index, item in enumerate(weapons):
+        weapons[index] = get_display_name(item)
+    for index, item in enumerate(ammunitions):
+        ammunitions[index] = get_display_name(item)
+    for index, item in enumerate(gears):
+        gears[index] = get_display_name(item)
+    for index, item in enumerate(mechs):
+        mechs[index] = get_display_name(item)
+    for index, item in enumerate(vehicles):
+        vehicles[index] = get_display_name(item)
+    for index, item in enumerate(battlearmors):
+        battlearmors[index] = get_display_name(item)
+    for index, item in enumerate(contracts):
+        contracts[index] = get_display_name(item)
+    
+    for index, item in enumerate(mechs):
+        mechs[index] = '#'.join(item.rsplit(' ', 1)) + '|' + item
+    for index, item in enumerate(vehicles):
+        vehicles[index] = '#'.join(item.rsplit(' ', 1)) + '|' + item
+
+    context = {
+    "faction_info": faction_info,
     "weapons": weapons,
     "ammunitions": ammunitions,
     "gears": gears,
     "mechs": mechs,
-    "mechparts": mechparts,
     "vehicles": vehicles, 
     "battlearmors": battlearmors,
     "contracts": contracts,
-}
-with open(results_filename, mode="w", encoding="utf-8") as results:
-    results.write(results_template.render(context))
-    print(f"... wrote {results_filename}")
+    }
+
+    with open(results_filename, mode="w", encoding="utf-8") as results:
+        results.write(template.render(context))
+        print(f"... wrote {results_filename}")
+
+def get_faction_specific_info(faction):
+    faction_lookup = {"Rasalhague": {"logo":"Rasalhague_logo.png", "name":"Free Rasalhague Republic", "link":"Free Rasalhague Republic"}}
+    return faction_lookup[faction]
+
+def get_display_name(item):
+    for root, dirs, files in os.walk(codebase_dir):
+        for file in files:
+            # Check if the file is the target JSON file
+            if file == item+".json":
```

Comment:
this is a pretty small function, but around this level of nesting is where I start to think about extracting a function.